src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -278,5 +279,9 @@ private ObjectTypeProperty UpdatePropertyDescription(ObjectTypeProperty property
                 property.ValueType,
                 property.OptionalViaNullability);
         }
+
+        private bool IsBaseClassWithDiscriminator => ObjectSchema.Discriminator?.All != null &&
+            ObjectSchema.Parents?.All.Count == 0 &&
+            !Configuration.MgmtConfiguration.SuppressAbstractBaseClass.Contains(ExistingType != null ? ExistingType.Name : DefaultName);
```

Comment:
This contains kind of complicated logic here, could we cache this value?
Is there a scenario that this value could change over time? If not, I suggest we cache this value or cache the `IsAbstract` property, so that this expression will only be invoked once.
And I have a bit concern that this configuration works on the generated type name instead of its "swagger name" - this introduces inconsistency with other configurations.
Please make sure this is well documented.

Reply:
I refactored this change, this part is reverted. And now the configuration will also use the swagger name.